src/amo/components/SearchForm.scss

```diff
@@ -1,48 +1,58 @@
 @import "~core/css/inc/vars";
+@import "~core/css/inc/mixins";
 
 $search-query-padding: 26px;
 
 .SearchForm-form {
   background: rgba(255, 255, 255, 0.2);
   border: 0;
-  display: flex;
 }
 
-.SearchForm-query {
+.SearchForm-query.CentredInput,
+.SearchForm-query .CentredInput-label,
+.SearchForm-query .CentredInput-input {
+  @include font-regular();
+  line-height: 3rem;
+  height: 3rem;
+}
+
+.SearchForm-query .CentredInput-label,
+.SearchForm-query .CentredInput-input {
+  font-size: $font-size-m;
+}
+
+.SearchForm-query .CentredInput-label {
+  color: rgba(255, 255, 255, 0.7);
+  text-overflow: ellipsis;
+}
+
+.SearchForm-query .CentredInput-label:before {
```

Comment:
I think this should be `::before`, since we aren't targetting any browsers that don't support the new notation.

Also, would be nice to nest.